contracts/libs/0.8.x/BytecodeStorage.sol

```diff
@@ -0,0 +1,257 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+// Created By: Art Blocks Inc.
+
+pragma solidity ^0.8.0;
+
+/**
+ * @title Art Blocks Script Storage Library
+ * @notice Utilize contract bytecode as persistant storage for large chunks of script string data.
+ *
+ * @author Art Blocks Inc.
+ * @author Modified from 0xSequence (https://github.com/0xsequence/sstore2/blob/master/contracts/SSTORE2.sol)
+ * @author Modified from Solmate (https://github.com/transmissions11/solmate/blob/main/src/utils/SSTORE2.sol)
+ *
+ * @dev Compared to the above two rerferenced libraries, this contracts-as-storage implementation makes a few
+ *      notably different design decisions:
+ *      - uses the `string` data type for input/output on reads, rather than speaking in bytes directly
+ *      - exposes "delete" functionality, allowing no-longer-used storage to be purged from chain state
+ *      - stores the "writer" address (library user) in the deployed contract bytes, which is useful for both:
+ *         a) providing necessary information for safe deletion; and
+ *         b) allowing this to be introspected on-chain
+ *      Also, given that much of this library is written in assembly, this library makes use of a slightly
+ *      different convention (when compared to the rest of the Art Blocks smart contract repo) around
+ *      pre-defining return values in order to simplify need to directly memory manage these return values.
+ */
+library BytecodeStorage {
+    //---------------------------------------------------------------------------------------------------------------//
+    // Offset Amount | Offset Aggregate | Description                                                                //
```

Comment:
Nit: I personally prefer documenting the starting offset and the size of the value, since those are the two most relevant parts for pointer math, but Solidity doesn't really do pointers so maybe that's not too important.

Reply:
I think I may be misunderstanding the feedback here, but in this case these offsets are refering to the deployed contract code that sits in storage vs. memory pointers/offsets, so the feedback may not exactly apply here? 

That said, perhaps we could do something like this, which I believe is in alignment with your nit here, if this seems clearer:

```
Starting Index | Size | Ending Index | Description
0                      | N/A  | 0                     | N/A, starting index is at 0.
0                      | 53    | 53                  | the bytes of the gated-cleanup-logic allowing for `selfdestruct`ion
53                    | 20    | 73                  | the 20 bytes used for storing the deploying-contract's address
```

WDYT @ryley-o @shantanubala ?

Reply:
That sgtm, but also not a huge deal, I was able to understand what was happening with the current documentation 👍

Reply:
@jakerockland i like your suggested code in your comment above. Seems more clear.

Reply:
I've updated the table to match @jakerockland 's suggested format above in this commit: https://github.com/ArtBlocks/artblocks-contracts/pull/299/commits/1f4e0a2d5a82e099e6b39685544e6d8fc9528059

Reply:
TYVM @ryley-o 🙏